api/routes/streaming.py

```
@@ -0,0 +1,76 @@
+    instead of an ``aclose()`` racing an in-flight pull.
+    """
+    queue: asyncio.Queue = asyncio.Queue()
+    finished = object()
+
+    async def _pump():
+        try:
+            async for chunk in chunks:
+                queue.put_nowait(chunk)
+        except asyncio.CancelledError:
+            raise
+        except BaseException as exc:  # pylint: disable=broad-exception-caught
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `BaseException`

(BLE001)